packages/backend/server/routes/[packageAndRefOrSha].get.ts

```diff
@@ -0,0 +1,42 @@
+import { WorkflowData } from "../types";
+
+type Params = Omit<WorkflowData, "sha" | "isPullRequest" | "ref"> & {
+  packageAndRefOrSha: string;
+};
+
+interface PackageData {
+  repository?: {
+    type: string;
+    url: string;
+  };
+}
+
+function extractOwnerAndRepo(repositoryUrl: string): [string, string] {
+  const url = new URL(repositoryUrl);
+  const parts = url.pathname.split("/").filter(Boolean);
+  if (parts.length >= 2) {
+    // first one is the owner, the second one is the repo name
+    return [parts[1], parts[2]];

```

Comment:
we can just write, return parts.slice(1, 3) or something like that. This does not look ideal.

One thing, you sure [1] & [2] are always owner & repo? have you tested it?